sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -106,9 +106,11 @@ case class InsertIntoHadoopFsRelationCommand(
         fs, catalogTable.get, qualifiedOutputPath, matchingPartitions)
     }
 
+    // For SPARK-27194 unit test, we try to set constant jobId carried by options
+    val jobId = options.getOrElse("test.jobId", java.util.UUID.randomUUID().toString)
```

Comment:
Can we try to reproduce the file collision without adding this extra option?

Reply:
good catch, I'll try to pre-create a partition file with a customized file commit protocol to keep this code clean